src/Hl7.Fhir.ElementModel/ElementNode.cs

```diff
@@ -176,6 +176,24 @@ public string Location
                     //TODO: Slow - but since we'll change the use of this property to informational 
                     //(i.e. for error messages), it may not be necessary to improve it.
                     var basePath = Parent.Location;
+                    var myIndex = Parent.ChildList.Where(c => c.Name == Name).ToList().IndexOf(this);
+                    return $"{basePath}.{Name}[{myIndex}]";
+
+                }
+                else
+                    return Name;
+            }
+        }
+
+        public string ShortPath
```

Comment:
Nice, did you also add:

```
  if (type == typeof(ElementNode) || type == typeof(ITypedElement) || type == typeof(IShortPathGenerator))
                return new[] { this };
```

to the `Annotations(Type type)` function?  Otherwise, there is no way to reach out to this ElementNode!